apps/dashboard/src/components/workflow-editor/steps/email/variables/variables.ts

```diff
@@ -1,49 +1,109 @@
-import { Variable, Variables } from '@maily-to/core/extensions';
-import type { Editor as TiptapEditor } from '@tiptap/core';
+import { IsAllowedVariable, LiquidVariable } from '@/utils/parseStepVariables';
+import type { Editor, Editor as TiptapEditor, Range } from '@tiptap/core';
+
+export const REPEAT_BLOCK_ITERABLE_ALIAS = 'current';
+
+export const ALLOWED_ALIASES = [REPEAT_BLOCK_ITERABLE_ALIAS];
 
 export enum VariableFrom {
-  Content = 'content-variable',
+  // variable coming from bubble menu (e.g. 'showIf')
   Bubble = 'bubble-variable',
-  Repeat = 'repeat-variable',
+  // variable coming from repeat block 'each' input
+  RepeatEachKey = 'repeat-variable',
+  // all the other variables
+  Content = 'content-variable',
 }
 
 export type CalculateVariablesProps = {
   query: string;
   editor: TiptapEditor;
   from: VariableFrom;
-  primitives: Array<Variable>;
-  arrays: Array<Variable>;
-  namespaces: Array<Variable>;
-  isAllowedVariable: (variable: string) => boolean;
+  primitives: Array<LiquidVariable>;
+  arrays: Array<LiquidVariable>;
+  namespaces: Array<LiquidVariable>;
+  isAllowedVariable: IsAllowedVariable;
   isEnhancedDigestEnabled: boolean;
 };
 
-function insertVariable({
+const insertNodeToEditor = ({
+  editor,
+  range,
+  nodeType,
+  nodeAttrs,
+}: {
+  editor: Editor;
+  range: Range;
+  nodeType: string;
+  nodeAttrs: Record<string, any>;
+}) => {
+  const nodeAfter = editor.view.state.selection.$to.nodeAfter;
+  const overrideSpace = nodeAfter?.text?.startsWith(' ');
+
+  // add space after variable if it's a text node
+  if (overrideSpace) {
+    range.to += 1;
+  }
+
+  editor
+    .chain()
+    .focus()
+    .insertContentAt(range, [
+      {
+        type: nodeType,
+        attrs: nodeAttrs,
+      },
+      {
+        type: 'text',
+        text: ' ',
+      },
+    ])
+    .run();
+};
+
+export const insertVariableToEditor = ({
```

Comment:
when I do type `current.payload.foo` manually it's not recognized as the `variable` but as the `text`

https://github.com/user-attachments/assets/6b1f488d-d6d5-4d07-8cc2-0e054f741f2d